File: python/examples/error_bars_xy.py
from __future__ import annotations

from _shared import ExampleMeta, base_plot, error_bar_xy_series, save_example

META = ExampleMeta(
    slug="error-bars-xy",
    title="Horizontal and vertical error bars",
    summary="A point series with uncertainty in both axes.",
    section="Statistical plots",
)


def build_plot():
    x, y, x_errors, y_errors = error_bar_xy_series()
    return (
        base_plot("XY Error Bars")
        .xlabel("throughput")
        .ylabel("latency")
        .error_bars_xy(x, y, x_errors, y_errors)
    )


if __name__ == "__main__":
    save_example(META, build_plot())

File: packages/ruviz/examples/error-bars-xy.ts
import { createPlot } from "../src/index.js";
import { errorBarXYSeries } from "./data.js";
import type { PlotExampleDefinition } from "./types.js";

export const example: PlotExampleDefinition = {
  slug: "error-bars-xy",
  title: "Horizontal and vertical error bars",
  summary: "A point series with uncertainty in both axes.",
  category: "statistical",
  createPlot() {
    const { x, y, xErrors, yErrors } = errorBarXYSeries();
    return createPlot()
      .sizePx(640, 320)
      .title("XY Error Bars")
      .xlabel("throughput")
      .ylabel("latency")
      .errorBarsXY({ x, y, xErrors, yErrors });
  },
};
